java/kotlin-extractor/src/main/kotlin/KotlinUsesExtractor.kt

```diff
@@ -716,7 +749,15 @@ open class KotlinUsesExtractor(
                 } else {
                     extractFileClass(dp)
                 }
-            is IrClass -> if (classTypeArguments != null && !dp.isAnonymousObject) useClassInstance(dp, classTypeArguments, inReceiverContext).typeResult.id else useClassSource(dp)
+            is IrClass ->
+                if (classTypeArguments != null && !dp.isAnonymousObject) {
+                    useClassInstance(dp, classTypeArguments, inReceiverContext).typeResult.id
+                } else {
+                    // `inReceiverContext == false` is used unless we have identified that we're dealing with a raw type
+                    // produced by the Parcelize plugin. In that case we're using the original `inReceiverContext`. Note
```

Comment:
If the class should really be the generic specialisation `List<Object>` or so, why aren't we going down the `useClassInstance` route?

Reply:
Good point. We could. I'll change this.